src/Adapter/Country/CommandHandler/AddCountryHandler.php

```diff
@@ -77,12 +87,8 @@ public function handle(AddCountryCommand $command): CountryId
                 $country->id_zone = $command->getZoneId()->getValue();
             }
 
-            $this->validateCountryFields($country);
-
-            if (false === $country->add()) {
-                throw new CannotAddCountryException('Failed to add country');
-            }
-        } catch (PrestaShopException $e) {
+            $this->countryRepository->add($country);
+        } catch (CoreException $e) {
```

Comment:
I don't think you need to catch this. If something goes wrong related to country, it will throw CannotAddCountry exception already. If it throws core exception then its something fatal, not related to country anymore (but rather smth with db connection etc...)